prospero-common/src/main/java/org/wildfly/prospero/actions/ApplyCandidateAction.java

```diff
@@ -278,12 +281,20 @@ public UpdateSet findUpdates() throws OperationException {
         }
         for (Artifact artifact : candidate) {
             candidateMap.put(artifact.getGroupId() + ":" + artifact.getArtifactId(), artifact);
+
         }
         List<ArtifactChange> changes = new ArrayList<>();
+
+        Map<String, String> channelMap = getChannelNameMap();
         for (String key : baseMap.keySet()) {
+
             if (candidateMap.containsKey(key)) {
                 if (!baseMap.get(key).getVersion().equals(candidateMap.get(key).getVersion())) {
-                    changes.add(ArtifactChange.updated(baseMap.get(key), candidateMap.get(key)));
+                    if (channelMap.isEmpty()) {
+                        changes.add(ArtifactChange.updated(baseMap.get(key), candidateMap.get(key)));
```

Comment:
I think you should add the change without the channel name if the `channelMap` doesn't contain the key as well. The `isEmpty` check is not really needed.